### PR TITLE
added GUI

### PR DESCRIPTION
**Pull request by rbral**

### CODE REVIEW

File: src/main/java/bral/dictionary/DictionaryComponent.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package bral.dictionary;
+
+import javax.swing.*;
+import java.awt.*;
+
+public class DictionaryComponent extends JComponent
```
Comment (andrewoid) [inline src/main/java/bral/dictionary/DictionaryComponent.java]: What is this for?

**Comment by rbral** [comment]
Screenshot 2024-04-09 173051

File: src/main/java/bral/dictionary/DefinitionsComponent.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
+package bral.dictionary;
+
+import javax.swing.*;
+import java.awt.*;
+import java.util.List;
+
+public class DefinitionsComponent extends JComponent
```
Comment (andrewoid) [inline src/main/java/bral/dictionary/DefinitionsComponent.java]: What is this class for?

File: src/main/java/bral/dictionary/DictionaryFrame.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+            List<String> currDefinitions = englishDictionary.getDefinition(currWord);
+            // definitionsComponent.setCurrDefinitions(currDefinitions);
+
+            // clear the old contents:
+            definitionsTextArea.setText("");
+            for (String definition : currDefinitions)
+            {
+                definitionsTextArea.append(definition + "\n\n");
+            }
+        } catch (NullPointerException e) {
```
Comment (andrewoid) [inline src/main/java/bral/dictionary/DictionaryFrame.java]: Why?

File: src/test/java/bral/dictionary/EnglishDictionaryTest.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    public void getDefinition() throws CsvValidationException, IOException {
+        // given
+        EnglishDictionary dictionary = new EnglishDictionary();
+
+        // when
+        List<String> definitions = dictionary.getDefinition("Aaronic");
+        String firstDef = definitions.get(0);
+
+        // then
+        assertEquals("Alt. of Aaronical", firstDef);
```
Comment (andrewoid) [inline src/test/java/bral/dictionary/EnglishDictionaryTest.java]: Assert all the definitions, not just one.

**Comment by andrewoid** [comment]
Add a Readme

File: src/main/java/bral/dictionary/DictionaryFrame.java (hunk 3 above)
Comment (rbral) [reply src/main/java/bral/dictionary/DictionaryFrame.java]: It won't work without try/catch since if the user types in a word that's not in the dictionary, it would freeze without this block

Comment (andrewoid) [reply src/main/java/bral/dictionary/DictionaryFrame.java]: Ok... But fix that without the try catch

File: src/main/java/bral/dictionary/DictionaryFrame.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    }
+
+    private void reDisplay()
+    {
+        // clear the old contents:
+        definitionsTextArea.setText("");
+
+        String currWord = searchField.getText();
+
+        if (englishDictionary.getDictionaryMap().containsKey(currWord))
```
Comment (andrewoid) [inline src/main/java/bral/dictionary/DictionaryFrame.java]: getDefinition() method should be all you need.

File: src/main/java/bral/dictionary/EnglishDictionary.java [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+/**
+ * Reads the englishDictionary file ONCE
+ */
+public class EnglishDictionary {
+    private final Map<String, List<String>> dictionaryMap;
+
+    public Map<String, List<String>> getDictionaryMap() {
+        return dictionaryMap;
+    }
```
Comment (andrewoid) [inline src/main/java/bral/dictionary/EnglishDictionary.java]: You can remove this method now

File: src/main/java/bral/dictionary/DictionaryFrame.java [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+
+        searchField = new JTextField("");
+        main.add(searchField, BorderLayout.NORTH);
+
+        definitionsTextArea = new JTextArea("Definitions will show here...");
+        definitionsTextArea.setEditable(false);
+        definitionsTextArea.setLineWrap(true);
+        definitionsTextArea.setWrapStyleWord(true);
+        main.add(definitionsTextArea, BorderLayout.CENTER);
```
Comment (andrewoid) [inline src/main/java/bral/dictionary/DictionaryFrame.java:41]: Why do you add this to the center?

File: src/main/java/bral/dictionary/DictionaryFrame.java [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+
+        JScrollPane scrollPane = new JScrollPane(definitionsTextArea);
+        main.add(scrollPane);
```
Comment (andrewoid) [inline src/main/java/bral/dictionary/DictionaryFrame.java]: You didn't answer my question and instead changed a different line that I wasn't even talking about.

Comment (rbral) [reply src/main/java/bral/dictionary/DictionaryFrame.java]: Thank you for the feedback. So I should remove the center for the definitionsTextArea? I'm not really sure what the problem is since it looks the same with or without it

Comment (andrewoid) [reply src/main/java/bral/dictionary/DictionaryFrame.java]: I didn't ask you to remove anything. 

I simply asked why you are adding it to the center. 

You just ignored the question. 

I'm asking you to explain the code you've written.

Comment (rbral) [reply src/main/java/bral/dictionary/DictionaryFrame.java]: Oh, I didn't realize I was supposed to explain, I just thought something was wrong with my code. I was using a lot of the ideas from our projectile GUI. I thought it would make sense to put the search field in the north and the textarea for the definitions field in the center so it would be one on top of the other. Just like we put different components in the west vs the center in the projectile project. But I guess it doesn't make a difference here because it's one on top of another and not separate sections like we had in projectile.

Comment (andrewoid) [reply src/main/java/bral/dictionary/DictionaryFrame.java]: Only one thing can be in the center. You put 2 things in the center. The second one replaces the first.

Comment (rbral) [reply src/main/java/bral/dictionary/DictionaryFrame.java]: Okay got it, thank you

File: src/main/java/bral/dictionary/DictionaryFrame.java [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+
+
+        searchField = new JTextField("");
+        main.add(searchField, BorderLayout.NORTH);
+
+        definitionsTextArea = new JTextArea("Definitions will show here...");
+        definitionsTextArea.setEditable(false);
+        definitionsTextArea.setLineWrap(true);
+        definitionsTextArea.setWrapStyleWord(true);
+        main.add(definitionsTextArea, BorderLayout.CENTER);
```
Comment (andrewoid) [inline src/main/java/bral/dictionary/DictionaryFrame.java]: Why did you remove the JScrollPane?

Comment (rbral) [reply src/main/java/bral/dictionary/DictionaryFrame.java]: I removed it because it was not doing anything I needed anymore. My original intent was so that the words would wrap around, but that was before I discovered the .setLineWrap, and now that is accomplished. But I can put it back I guess if there are words with a lot of definitions

**Review by andrewoid — APPROVED** [review]

**Review by andrewoid — APPROVED** [review]
